core/src/main/java/org/frankframework/align/content/JsonDocumentContainer.java

```diff
@@ -91,12 +98,103 @@ protected void addContent(JsonElementContainer parent, JsonElementContainer chil
 		parent.addContent(child);
 	}
 
+	/**
+	 * Produce a {@link Message} with the JSON Document content, using {@link MessageBuilder} to reduce memory usage.
+	 */
+	public Message toMessage() throws IOException {
+		return toMessage(true);
+	}
+
+	/**
+	 * Produce a {@link Message} with the JSON Document content, using {@link MessageBuilder} to reduce memory usage.
+	 */
+	public Message toMessage(boolean indent) throws IOException {
+		if (root.getContent() == null) {
+			return Message.nullMessage(new MessageContext().withMimeType(MediaType.APPLICATION_JSON));
+		}
+		MessageBuilder messageBuilder = new MessageBuilder();
+		messageBuilder.setMimeType(MediaType.APPLICATION_JSON);
+		try (Writer writer = messageBuilder.asWriter()) {
+			toWriter(writer, indent);
+		}
+		return messageBuilder.build();
+	}
+
+	/**
+	 * Write JSON document content to the given {@link Writer}.
+	 *
+	 * @param writer
+	 * @param indent
+	 * @throws IOException
+	 */
+	public void toWriter(Writer writer, boolean indent) throws IOException {
+		Object content = root.getContent();
+		if (content == null) {
+			return;
+		}
+		if (skipRootElement && content instanceof Map<?, ?> map) {
+			content = map.values().toArray()[0];
+		}
+		toWriter(writer, content, indent ? 0 : -1);
+	}
+
+	protected void toWriter(Writer w, Object item, int indentLevel) throws IOException {
```

Comment:
Ben je niet hier de JsonWriter opnieuw aan het uitvinden?

Reply:
Hm, ik weet het niet. Ik heb de bestaande code genomen die alles in een StringBuffer stopt, en die code met zo min mogelijk aanpassingen overgezet om naar een Writer te schrijven ipv die StringBuffer.

Ik had er daarbij niet eens stilgestaan dat er al een JsonWriter is; ik zal daar even naar kijken.

Reply:
@nielsm5 Ik heb er even goed naar gekeken maar ik zou dit nu eigenlijk niet willen meenemen.
Ik zou dan de JsonWriter een optie willen geven om de JSON output te indenteren en formatteren, zoals de JsonDocumentContainer nu doet. Dat maakt de output wel zo leesbaar.

En eigenlijk wil ik de hele JsonDocumentContainer weg doen en direct vanuit de Xml2Json naar de JsonWriter schrijven, daarvoor heb ik al een TODO in de code staan. (Ik moet er wel nog een ticket voor maken).

Dan doe ik liever dat samen in een nieuw PR, zodat de change niet nog groter gaat worden en het niet nog langer gaat duren voordat dit gemerged kan worden.

Als je het liever wel meteen meeneemt, doe ik het natuurlijk.

Reply:
Prima, in een nieuw pr kan.